addons/website_sale/views/templates.xml

```diff
@@ -660,12 +661,15 @@
                 <input type="hidden" t-if="len(filtered_sorted_variants) == 1" class="product_id" name="product_id" t-att-value="filtered_sorted_variants[0].id"/>
                 <t t-if="len(filtered_sorted_variants) &gt; 1">
                     <div t-foreach="filtered_sorted_variants" t-as="variant_id" class="custom-control custom-radio">
-                        <input type="radio" name="product_id" class="custom-control-input product_id js_product_change" t-att-checked="'checked' if variant_id_index == 0 else None" t-attf-id="radio_variant_#{variant_id.id}" t-att-value="variant_id.id" t-att-data-lst_price="compute_currency(variant_id.lst_price)" t-att-data-price="variant_id.price" t-att-data-combination="variant_id.product_template_attribute_value_ids.ids"/>
+                        <t t-set="combination" t-value="variant_id.product_template_attribute_value_ids"/>
```

Comment:
why not just add this in  def product(self, product... ?

Reply:
It is view logic, if the view is called from elsewhere, we would have to copy/paste the line as many time as it is called.

Reply:
![image](https://user-images.githubusercontent.com/4848797/52116984-e1bc6d80-2612-11e9-8cba-e0a5c5e09573.png)

Deja que l'on a tendance a mettre bcp de la logique dans les controllers :/

Si on pouvait éviter d'en mettre dans les vues ;)

Celui qui voudra de toute facon faire un t-call de products, devra deja presetter plein de truc, et meme des fonctions :/

values = {
            'search': search,
            'category': category,
            'pricelist': pricelist,
            'attrib_values': attrib_values,
            # compute_currency deprecated, get from product
            'compute_currency': compute_currency,
            'attrib_set': attrib_set,
            'keep': keep,
            'categories': categs,
            'main_object': product,
            'product': product,
            'add_qty': add_qty,
            'optional_product_ids': [p.with_context({'active_id': p.id}) for p in product.optional_product_ids],
            # get_attribute_exclusions deprecated, use product method
            'get_attribute_exclusions': self._get_attribute_exclusions,
        }


Donc non, on ne part pas du principe de mettre un max de logique dans la couche View.